sirepo/template/template_common.py

```diff
@@ -83,6 +85,54 @@ def __init__(self, **kwargs):
         )
 
 
+class LogParser:
+    def __init__(
+        self,
+        run_dir,
+        log_filename,
+        error_patterns=_DEFAULT_ERROR_PATTERNS,
+        multiline_error_pattern=r"^Traceback .*?^\w*Error: (.*?)\n",
+        default_msg="An unknown error occurred",
+    ):
+        self.run_dir = run_dir
```

Comment:
We tend to inherit from PKDict to avoid having this repetition:

```python
class LogParser(PKDict):
  def __init__(self, run_dir, log_filename**kwargs):
    super().__init__(run_dir=run_dir, log_filename=log_filename, **kwargs)
    self.pksetdefault(
      error_patterns=_DEFAULT_ERROR_PATTERNS,
       ...
      )